web/src/narrow_hooks.ts

```diff
@@ -0,0 +1,68 @@
+type Hook = () => void;
+type Hook_ = (param: any) => void;
```

Comment:
Do not use `any`. This is hiding all the places where you unsafely convert from functions that don’t take arguments to functions that do; also, we never want to disable type checking for the argument.

Reply:
Yes, thank you for the advice! I understand this totally so this is actually a WIP - just want to see if this is a right direction. I was planning to write a precise type annotaiton later.